include/capypdf.hpp

```diff
@@ -96,6 +100,54 @@ template<typename T> class CapyC {
     std::unique_ptr<T, CapyCTypeDeleter> _d;
 };
 
+class Destination : public CapyC<CapyPDF_Destination> {
+public:
+    friend class LinkAnnotation;
+
+    Destination() {
+        CapyPDF_Destination *dest;
+        CAPY_CPP_CHECK(capy_destination_new(&dest));
+        _d.reset(dest);
+    }
+
+    void set_page_fit(int32_t page_num) {
+        CAPY_CPP_CHECK(capy_destination_set_page_fit(*this, page_num));
+    }
+
+    void set_page_xyz(int32_t page_num,
+                      std::optional<double> x,
+                      std::optional<double> y,
+                      std::optional<double> z) {
+        CAPY_CPP_CHECK(capy_destination_set_page_xyz(
+            *this, page_num, x ? &*x : nullptr, y ? &*y : nullptr, z ? &*z : nullptr));
+    }
+};
+
+class LinkAnnotation : public CapyC<CapyPDF_Annotation> {
```

Comment:
This should be `Annotation` not `LinkAnnotation`.

The underlying problem here being that the class hierarchy of annotations is not fully fleshed out. I made annotations just feature rich enough to deal with basic cases and then moved on to do more important work. What probably should happen is that annotations work like drawing contexts in that they all have the same type and calling a method that is not meaningful for a given annotation type is a run time error.

Reply:
I've fixed it for the cpp API, although the constructor is kept as a link annotation, I assume that will change in the C api at some point and I don't have to worry at the present time to support file or text annotations.

Reply:
I'm guessing that it needs to either change so that you construct them via the generator like so:

    auto annotation = gen.new_link_annotation(...);

or that it becomes a static constructor method in the class like so:

    auto annotation = Annotation::new_link_annotation(...);

In any case this piece of the API will change at some point or another.